### PR TITLE
feat(weave): Implement VertexAI integration

### PR DESCRIPTION
**Pull request by soumik12345**
# Description

This PR adds the autopatch integration with [VertexAI Generative Models API](https://cloud.google.com/vertex-ai/generative-ai/docs/start/quickstarts/quickstart-multimodal).

# Supported cases

### Without Streaming

```python
import vertexai
import weave
from vertexai.generative_models import GenerativeModel

weave.init(project_name="google_ai_studio-test")
vertexai.init(project="wandb-growth", location="us-central1")
model = GenerativeModel("gemini-1.5-flash-002")
response = model.generate_content(
    "What's a good name for a flower shop specialising in selling dried flower bouquets?"
)
```

[**Sample trace**](https://wandb.ai/geekyrakshit/google_ai_studio-test/weave/calls/0192b079-9872-7af3-bed0-f916f1084a6c)

### With Streaming

```python
import vertexai
import weave
from vertexai.generative_models import GenerativeModel

weave.init(project_name="google_ai_studio-test")
vertexai.init(project="wandb-growth", location="us-central1")
model = GenerativeModel("gemini-1.5-flash-002")
response = model.generate_content(
    "What's a good name for a flower shop specialising in selling dried flower bouquets?",
    stream=True,
)
```

[**Sample trace**](https://wandb.ai/geekyrakshit/google_ai_studio-test/weave/calls/0192b07a-c6ca-79f3-a059-53801c519e97)

### Without Streaming

```python
import vertexai
import weave
from vertexai.generative_models import GenerativeModel

weave.init(project_name="google_ai_studio-test")
vertexai.init(project="wandb-growth", location="us-central1")
model = GenerativeModel("gemini-1.5-flash-002")

async def async_generate():
    response = await model.generate_content_async(
        "What's a good name for a flower shop that specializes in selling bouquets of dried flowers?"
    )
    return response

response = asyncio.run(async_generate())
```

[**Sample trace**](https://wandb.ai/geekyrakshit/google_ai_studio-test/weave/calls/0192b07a-1ff7-7ab0-9666-a4942be7f244)

### With Streaming

```python
import vertexai
import weave
from vertexai.generative_models import GenerativeModel

weave.init(project_name="google_ai_studio-test")
vertexai.init(project="wandb-growth", location="us-central1")
model = GenerativeModel("gemini-1.5-flash-002")

async def get_response():
    chunks = []
    async for chunk in await model.generate_content_async(
        "What's a good name for a flower shop that specializes in selling bouquets of dried flowers?",
        stream=True,
    ):
        if chunk.text:
            chunks.append(chunk.text)
    return chunks

response = asyncio.run(get_response())
```

[**Sample trace**](https://wandb.ai/geekyrakshit/google_ai_studio-test/weave/calls/0192b07b-8380-7202-86dd-59f614d72bd6)

### CODE REVIEW

File: tests/integrations/vertexai/vertexai_test.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    calls = list(client.calls())
+    assert len(calls) == 1
+
+    call = calls[0]
+    assert call.started_at < call.ended_at
+
+    trace_name = op_name_from_ref(call.op_name)
+    assert trace_name == "vertexai.GenerativeModel.generate_content"
+    assert call.output is not None
```
Comment (andrewtruong) [inline tests/integrations/vertexai/vertexai_test.py:24]: can you make a more relevant assert using a mock?  This is just checking if anything populated at all.

File: tests/integrations/vertexai/vertexai_test.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
```
Comment (andrewtruong) [inline tests/integrations/vertexai/vertexai_test.py:49]: same here

File: tests/integrations/vertexai/vertexai_test.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+    calls = list(client.calls())
+    assert len(calls) == 1
+
+    call = calls[0]
+    assert call.started_at < call.ended_at
+
+    trace_name = op_name_from_ref(call.op_name)
+    assert trace_name == "vertexai.GenerativeModel.generate_content_async"
+    assert call.output is not None
```
Comment (andrewtruong) [inline tests/integrations/vertexai/vertexai_test.py:71]: same here

File: tests/integrations/vertexai/vertexai_test.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
```
Comment (andrewtruong) [inline tests/integrations/vertexai/vertexai_test.py:103]: same here

File: tests/integrations/vertexai/vertexai_test.py (hunk 1 above)
Comment (soumik12345) [reply tests/integrations/vertexai/vertexai_test.py]: I added cassettes for `test_content_generation` and `test_content_generation_stream`. However, I'm unable to generate them for `test_content_generation_async` and `test_content_generation_async_stream`.

Comment (andrewtruong) [reply tests/integrations/vertexai/vertexai_test.py]: Do cassettes work here?  I thought this used the google grpc stuff under the hood?

In the case where they don't work, you (or gpt!) will need to mock out the request/response manually

Comment (soumik12345) [reply tests/integrations/vertexai/vertexai_test.py]: @andrewtruong Do you mean that we need to write the cassettes manually for those specific functions? Or is there another way?

**Comment by andrewtruong** [comment]
It's unfortunate we can't use vcrpy here.  I think you'll need to manually mock the Vertex calls

**Comment by andrewtruong** [comment]
It's unfortunate we can't use VCRpy here.  Instead of mocking, you can put a 5x retry on the test.  It's not great, but given how fast these things move it might be the best option

**Comment by andrewtruong** [comment]
Your pytest markers should be called `flaky` -- see example: https://github.com/wandb/weave/blame/master/tests/integrations/google_ai_studio/google_ai_studio_test.py#L50

**Review by andrewtruong — APPROVED** [review]
Approving with the understanding that you have thoroughly checked the integration since these tests are skipped

**Pull request merged**